src/components/JitsiMeeting.tsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useRef, useState } from "react";
+import { init } from "..";
+import { IJitsiMeetingProps } from "../types";
+import { getRoomName } from "../utils";
+
+/**
+ * Returns the JitsiMeeting Component with access to a custom External API
+ * to be used as-it-is in React projects
+ *
+ * @param {IJitsiMeetingProps} props
+ * @example
+  ```js
+      <JitsiMeeting
+        id="testId"
+        domain="meet.jit.si"
+        options={{
+          roomName: 'TestingJitsiMeetingComponent'
+          width: '100%'
+          height: '500px'
+        }}
+        spinner={() => <>loading jitsi meeting...</>}
+        onApiReady={(externalApi) => console.log(externalApi)}
+      />
+  ```
+ */
+const JitsiMeeting = ({
+  id,
+  domain,
+  options,
+  tenant,
+  spinner,
```

Comment:
it would seem that the `spinner` prop is expected to be directly a react component rather than a wrapper function over a react component. Maybe it would be better to pass it that way? wdyt?